myst_nbs/time_series/Forecasting_with_structural_timeseries.myst.md

```diff
@@ -0,0 +1,747 @@
+        y=idata_ar["constant_data"]["y"],
+        color="k",
+        label="Observed Data points",
+    )
+    axs[0].set_title("Prior Predictive Fit", fontsize=20)
+    axs[0].legend()
+
+    for i, p in enumerate(percs[::-1]):
+        upper = np.percentile(
+            az.extract_dataset(idata_preds, group="predictions", num_samples=100)["likelihood"],
+            p,
+            axis=1,
+        )
+        lower = np.percentile(
+            az.extract_dataset(idata_preds, group="predictions", num_samples=100)["likelihood"],
+            100 - p,
+            axis=1,
+        )
+        color_val = colors[i]
+        axs[2].fill_between(
+            x=idata_preds["predictions_constant_data"]["t"],
+            y1=upper.flatten(),
+            y2=lower.flatten(),
+            color=cmap(color_val),
+            alpha=0.1,
+        )
+
+        upper = np.percentile(
+            az.extract_dataset(idata_preds, group="predictions", num_samples=100)["yhat_fut"],
+            p,
+            axis=1,
+        )
+        lower = np.percentile(
+            az.extract_dataset(idata_preds, group="predictions", num_samples=100)["yhat_fut"],
+            100 - p,
+            axis=1,
+        )
+        color_val = colors[i]
+        axs[2].fill_between(
+            x=np.linspace(200, 250, 50),
+            y1=upper.flatten(),
+            y2=lower.flatten(),
+            color=cmap(color_val),
+            alpha=0.1,
+        )
+
+    axs[2].plot(
+        az.extract_dataset(idata_preds, group="predictions", num_samples=100)["likelihood"].mean(
+            axis=1
+        ),
+        color="cyan",
+    )
+    idata_preds.predictions.yhat_fut.mean(["chain", "draw"]).plot(
+        ax=axs[2], color="cyan", label="Predicted Mean Realisation"
+    )
+    axs[2].scatter(
+        x=idata_ar["constant_data"]["t"],
+        y=idata_ar["constant_data"]["y"],
+        color="k",
+        label="Observed Data",
+    )
+    axs[2].set_title("Posterior Predictions Plotted", fontsize=20)
+    axs[2].axvline(np.max(idata_ar["constant_data"]["t"]), color="black")
+    axs[2].legend()
+    axs[2].set_xlabel("Time in Days")
+    axs[0].set_xlabel("Time in Days")
+    az.plot_ppc(idata_ar, ax=axs[1])
+
+
+plot_fits(idata_ar, idata_preds)
+```
+
+Here we can see that although the model converged and ends up with a reasonable fit to the existing the data, and a **plausible  projection** for future values. However, we have set the prior specification very poorly in allowing an absurdly broad range of values due to the kind of compounding logic of the auto-regressive function. For this reason it's very important to be able to inspect and tailor your model with prior predictive checks. 
+
+Secondly, the mean forecast fails to capture any long lasting structure, quickly dying down to a stable baseline. To account for these kind of short-lived forecasts, we can add more structure to our model, but first, let's complicate the picture. 
```

Comment:
More of a stylistic suggestion, feel free to ignore. At the moment the wording just suggests increased complexity for the sake of it. Suggest rephrasing to motivate _why_ we might want to add extra model structure... to improve the model in specific ways.

Reply:
Makes sense.